frappe/utils/__init__.py

```diff
@@ -66,9 +66,14 @@ def get_email_address(user=None):
 def get_formatted_email(user, mail=None):
 	"""get Email Address of user formatted as: `John Doe <johndoe@example.com>`"""
 	fullname = get_fullname(user)
+
+	if not mail:
+		mail = get_email_address(user) or validate_email_address(user)
```

Comment:
`validate_email_address` takes `email_str` which is what you'd get from `get_email_address`. I'm not sure if email needs to be validated here. If email does need to be validated, maybe email should be validated even when the `mail` parameter exists.

Reply:
So `get_email_address` returns value from `tabUser` table. 

Here is the case, that email id does not belong to `tabUser`, thus validating `user` value using `validate_email_address` method.